temporalio/workflow.py

```diff
@@ -1404,20 +1407,32 @@ def _apply_to_class(
                             f"@workflow.update defined on {base_member.__qualname__} but not on the override"
                         )
 
+        init_fn_takes_workflow_input = False
+
         if not seen_run_attr:
             issues.append("Missing @workflow.run method")
-        if len(issues) == 1:
-            raise ValueError(f"Invalid workflow class: {issues[0]}")
-        elif issues:
+        if not init_fn:
+            issues.append("Missing __init__ method")
```

Comment:
Thanks, very helpful point. I've changed the implementation to defer making the decision from import time to workflow-start time; IMO your counter-example clearly shows that defering is the only option. And I've switched this PR to Draft state because I think we probably do need to require the decorator.